plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -292,13 +293,96 @@ export class EventsProcessor {
             return
         }
         if (event === '$create_alias') {
-            await this.alias(properties['alias'], distinctId, teamId, timestamp, false)
+            await this.merge(properties['alias'], distinctId, teamId, timestamp, false)
         } else if (event === '$identify' && properties['$anon_distinct_id']) {
-            await this.alias(properties['$anon_distinct_id'], distinctId, teamId, timestamp)
+            await this.merge(properties['$anon_distinct_id'], distinctId, teamId, timestamp, true)
         }
     }
 
-    private async alias(
+    private async merge(
+        previousDistinctId: string,
+        distinctId: string,
+        teamId: number,
+        timestamp: DateTime,
+        isIdentifyCall: boolean
+    ): Promise<void> {
+        // No reason to alias person against itself. Done by posthog-js-lite when updating user properties
+        if (distinctId === previousDistinctId) {
+            return
+        }
+        if (this.isNewPersonPropertiesUpdateEnabled(teamId)) {
+            await this.mergeNew(distinctId, previousDistinctId, teamId, timestamp)
+        } else {
+            await this.aliasDeprecated(previousDistinctId, distinctId, teamId, timestamp, isIdentifyCall)
+        }
+    }
+
+    private async mergeNew(
+        dId1: string,
```

Comment:
Let's use more readable variable names.

`distinctId1` is not long and much nicer

Reply:
I find it hard to distinguish 1 vs 2 and the longer the name the harder it gets, any other naming suggestions? `oldId` `newId` or something similar would be great, but I couldn't figure out what would actually apply here as they are pretty equal now.